src/views/bridge-confirmation/bridge-confirmation.view.tsx

```diff
@@ -77,102 +69,55 @@ const BridgeConfirmation: FC = () => {
   const [estimatedGas, setEstimatedGas] = useState<AsyncTask<Gas, string>>({
     status: "pending",
   });
-  const estimateGasPollInterval = useRef<number>();
+  const [fadeClass, setFadeClass] = useState<string>();
   const currencySymbol = getCurrencySymbol(getCurrency());
 
-  const startGasPolling = useCallback(() => {
-    const estimateGas = () => {
-      setEstimatedGas((currentEstimatedGas) =>
-        isAsyncTaskDataAvailable(currentEstimatedGas)
-          ? { status: "reloading", data: currentEstimatedGas.data }
-          : { status: "loading" }
-      );
-      if (formData && connectedProvider.status === "successful" && tokenBalance) {
+  useEffect(() => {
+    if (connectedProvider.status === "successful" && formData && tokenBalance) {
+      const onBlock = () => {
         const { from, to, token, amount } = formData;
         const destinationAddress = connectedProvider.data.account;
 
-        estimateBridgeGas({
+        console.log("Pasa");
+        setEstimatedGas(
+          estimatedGas.status === "successful"
+            ? { status: "reloading", data: estimatedGas.data }
+            : { status: "loading" }
+        );
```

Comment:
This should also be reverted to the original code:

```typescript
setEstimatedGas((oldEstimatedGas) =>
  oldEstimatedGas.status === "successful"
    ? { status: "reloading", data: oldEstimatedGas.data }
    : { status: "loading" }
);
```

We can not have `estimatedGas` as dependency of this effect.

Reply:
Right, fixed!

Reply:
> We can not have estimatedGas as dependency of this effect.

Why not?

Reply:
This is related to calling `onBlock();` for the initial request. In any case, you already fixed it like this, which is ok:

```typescript
if (estimatedGas.status === "pending") {
  onBlock();
}
```